src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -177,6 +255,60 @@ public async Task<Stream> GetFileFrameAsync(long version, FrameRange range, Canc
         return stream;
     }
 
+    /// <inheritdoc />
+    public async Task<BinaryData> GetFileContentInRangeAsync(long version, FrameRange range, CancellationToken cancellationToken)
+    {
+        EnsureArg.IsNotNull(range, nameof(range));
+
+        BlockBlobClient blob = GetInstanceBlockBlobClient(version);
+
+        BinaryData data = null;
+        var blobDownloadOptions = new BlobDownloadOptions
+        {
+            Range = new HttpRange(range.Offset, range.Length)
+        };
+
+        await ExecuteAsync(async () =>
+        {
+            Response<BlobDownloadResult> result = await blob.DownloadContentAsync(blobDownloadOptions, cancellationToken);
+            data = result.Value.Content;
+        });
+
+        return data;
+    }
+
+    /// <inheritdoc />
+    public async Task<KeyValuePair<string, long>> GetFirstBlockPropertyAsync(long version, CancellationToken cancellationToken = default)
+    {
+        BlockBlobClient blobClient = GetInstanceBlockBlobClient(version);
+        KeyValuePair<string, long> result = new KeyValuePair<string, long>();
+
+        await ExecuteAsync(async () =>
+        {
+            BlockList blockList = await blobClient.GetBlockListAsync(BlockListTypes.Committed, snapshot: null, conditions: null, cancellationToken);
+
+            if (!blockList.CommittedBlocks.Any())
+                throw new DataStoreException(DicomBlobResource.BlockListNotFound);
+
+            BlobBlock firstBlock = blockList.CommittedBlocks.First();
+            result = new KeyValuePair<string, long>(firstBlock.Name, firstBlock.Size);
+        });
+
+        return result;
+    }
+
+    /// <inheritdoc />
+    public async Task CopyFileAsync(long originalVersion, long newVersion, CancellationToken cancellationToken)
```

Comment:
Good thought, I am thinking of throwing an exception if the copy hasnt completed.

Reply:
Presumably this is here in case we re-run the activity, right? So perhaps it'd be nice to log a message saying it already exists and we're skipping inside of the activity?